connect_ext_test.go

```diff
@@ -543,6 +543,51 @@ func TestTimeoutParsing(t *testing.T) {
 	assert.Nil(t, err)
 }
 
+func TestFailCodec(t *testing.T) {
+	t.Parallel()
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	server := httptest.NewServer(handler)
+	defer server.Close()
+	client := pingv1connect.NewPingServiceClient(
+		server.Client(),
+		server.URL,
+		connect.WithCodec(failCodec{}),
+	)
+	stream := client.CumSum(context.Background())
+	err := stream.Send(&pingv1.CumSumRequest{})
+	var connectErr *connect.Error
+	assert.NotNil(t, err)
+	assert.True(t, errors.As(err, &connectErr))
+	assert.Equal(t, connectErr.Code(), connect.CodeInternal)
+}
+
+func TestContextError(t *testing.T) {
+	t.Parallel()
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {})
+	server := httptest.NewServer(handler)
+	defer server.Close()
+	client := pingv1connect.NewPingServiceClient(
+		server.Client(),
+		server.URL,
+	)
+	wantErr := error(connect.NewError(connect.CodeCanceled, errors.New("fail")))
+	ctx := errorContext{Context: context.Background(), Error: wantErr}
+	stream := client.CumSum(ctx)
+	err := stream.Send(nil)
+	var connectErr *connect.Error
+	assert.NotNil(t, err)
+	assert.True(t, errors.As(err, &connectErr))
+	assert.Equal(t, connectErr.Code(), connect.CodeCanceled)
+
+	wantErr = errors.New("fail")
+	ctx = errorContext{Context: context.Background(), Error: wantErr}
```

Comment:
This is where we'd probably want to test deadline exceeded (in addition to canceled). The context type is an interface but it is documented to either return Canceled or DeadlineExceeded: https://pkg.go.dev/context#Context